### PR TITLE
Fix parameter type mapping for Sqlite

### LINKED ISSUE #27213: Correct SqliteType inference from string, DateOnly, TimeOnly
Microsoft.Data.Sqlite 6.0.1
TargetFramework net6.0-windows

当我创建一个字符串类型的SqliteParameter时，我查询它的类型，得到的类型信息是 “SqliteType.Integer”

```C#

var p = new Microsoft.Data.Sqlite.SqliteParameter("Name","123");

```

后来我在源码中定位到了以下位置，这就有点不合适了吧

https://github.com/dotnet/efcore/blob/8144d50ec0d15b87c35a9a3a425b1faadafb87f1/src/Microsoft.Data.Sqlite.Core/SqliteValueBinder.cs#L240-L263

### PR DESCRIPTION
**Pull request by Marusyk**
Small fix of the parameter type mapping

Fixes #27213

### CODE REVIEW

**Comment by Saibamen** [comment]
Unit tests needed?

**Comment by bricelam** [comment]
Rebased and added some tests.

@dotnet/efteam Can I get a second review since I changed additional product code?

**Review by smitpatel — APPROVED** [review]

File: test/Microsoft.Data.Sqlite.Tests/SqliteParameterTest.cs [hunk 1]
```diff
@@ -93,6 +93,68 @@ public void SqliteType_defaults_to_text()
...
+        [Fact]
+        public void SqliteType_is_inferred_from_Guid_value()
+            => SqliteType_is_inferred_from_value(default(Guid), SqliteType.Text);
+
+        [Fact]
+        public void SqliteType_is_inferred_from_TimeSpan_value()
+            => SqliteType_is_inferred_from_value(default(TimeSpan), SqliteType.Text);
+
+        [Fact]
+        public void SqliteType_overrides_inferred_value()
```
Comment (roji) [inline test/Microsoft.Data.Sqlite.Tests/SqliteParameterTest.cs:152]: Probably better to collapse all these to a single method and use [MemberData].

**Review by roji — APPROVED** [review]

**Comment by daniel-smith** [comment]
It looks like this has been approved for 3 months. Any ideas on when this will be merged?

**Comment by bricelam** [comment]
I mean, I have til November, don't I? 😏

**Comment by bricelam** [comment]
Thanks for you contribution! (lol, and patience)

**Pull request merged**